packages/grafana-ui/src/components/Button/styles.ts

```diff
@@ -133,8 +133,7 @@ export const getButtonStyles = stylesFactory(({ theme, size, variant, withIcon }
     `,
     icon: css`
       label: button-icon;
-      margin-right: ${iconDistance};
-      filter: brightness(100);
+      margin-right: ${withText ? iconDistance : 0};
```

Comment:
Instead of using this can we apply margin left to the span that's followed by the icon in ButtonContent using + selector maybe? then we could probably get rid of `withIcon` and `withText` in `getButtonStyles`

Reply:
But we can have button without icon. So still would have to use that margin conditionally (though would not need a new variable). But realised we may not need margin, maybe it would be better to use just empty non braking space here. That way the spacing would be font size dependant which makes more sense to me.

Reply:
What i'm thinking abut is:

```
.contentSpan {
  margin-left: 0;
}
.icon {
  + .contentSpan {
    margin-left: XXX
  }
}
```

Which boils down to always rendering content within a span, which is fine.

Reply:
it's a simple one, no worries :)